agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -3622,23 +3622,23 @@ bool slave_thread::send_cmdu_to_controller(ieee1905_1::CmduMessageTx &cmdu_tx)
  *        class member params authkey and keywrapauth are computed
  *        on success.
  *
- * @param[in] m1 WSC M1 TLV received from the radio agent
+ * @param[in] m2 WSC M2 recived from the controller
  * @param[out] authkey 32 bytes calculated authentication key
  * @param[out] keywrapkey 16 bytes calculated key wrap key
  * @return true on success
  * @return false on failure
  */
-bool slave_thread::autoconfig_wsc_calculate_keys(std::shared_ptr<ieee1905_1::tlvWscM2> m2,
-                                                 uint8_t authkey[32], uint8_t keywrapkey[16])
+bool slave_thread::autoconfig_wsc_calculate_keys(WSC::m2 &m2, uint8_t authkey[32],
+                                                 uint8_t keywrapkey[16])
 {
     auto mac = network_utils::mac_from_string(backhaul_params.bridge_mac);
```

Comment:
No need to "waste" time on converting the MAC address if the DH member is not initialized